components/sup/src/manager/service_updater.rs

```diff
@@ -102,8 +104,46 @@ impl ServiceUpdater {
         }
     }
 
-    // TODO (CM): How do we remove something from the updater? e.g.,
-    // when we stop or unload a service?
+    /// Remove a `Service` from updates, e.g. if the service was unloaded.
+    pub fn remove(&mut self, service: &Service) {
+        let updater_state = self.states.remove(&service.service_group);
+
+        if updater_state.is_none() {
+            warn!(
+                "Tried to remove {} from the ServiceUpdater, but it wasn't found.",
+                service
+            );
+            return;
+        }
+
+        match updater_state {
+            Some(UpdaterState::AtOnce(_rx, kill_tx)) => {
+                if kill_tx.send(true).is_err() {
+                    debug!("Tried to kill the updater thread but it's already dead.");
+                }
+            }
+            Some(UpdaterState::Rolling(rs)) => match rs {
+                RollingState::Leader(ls) => match ls {
+                    LeaderState::Polling(_rx, kill_tx) => {
+                        if kill_tx.send(true).is_err() {
+                            debug!("Tried to kill the updater thread but it's already dead.");
+                        }
+                    }
+                    LeaderState::Waiting => {}
+                },
+                RollingState::Follower(fs) => match fs {
+                    FollowerState::Updating(_rx, kill_tx) => {
+                        if kill_tx.send(true).is_err() {
+                            debug!("Tried to kill the updater thread but it's already dead.");
+                        }
+                    }
+                    FollowerState::Waiting => {}
+                },
+                _ => {}
+            },
+            None => {}
```

Comment:
This is handled above on line 111... might as well consolidate the entire body of this function into one big `match` statement.

Reply:
Good catch